llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7972,6 +7972,13 @@ Instruction *InstCombinerImpl::visitFCmpInst(FCmpInst &I) {
   Constant *RHSC;
   if (match(Op0, m_Instruction(LHSI)) && match(Op1, m_Constant(RHSC))) {
     switch (LHSI->getOpcode()) {
+    case Instruction::Select:
+      if ((Pred == FCmpInst::FCMP_UEQ || Pred == FCmpInst::FCMP_OEQ ||
+           Pred == FCmpInst::FCMP_UNE || Pred == FCmpInst::FCMP_ONE) &&
+          match(LHSI, m_Select(m_Value(), m_Value(X), m_FNeg(m_Value(Y)))) &&
+          X == Y && match(RHSC, m_AnyZeroFP()))
```

Comment:
```suggestion
          match(LHSI, m_Select(m_Value(), m_Value(X), m_FNeg(m_Specific(X)))) && match(RHSC, m_AnyZeroFP()))
```

Reply:
wouldn't it be better if it's `m_OneUse(m_FNeg(m_Deferred(X)))`?

Reply:
No - the transform is still beneficial even if we can't remove the fneg instruction.